src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/endpoints/ModelConverterAction.java

```diff
@@ -142,6 +148,105 @@ public HttpResponse doToJson(StaplerRequest req) {
 
     @SuppressWarnings("unused")
     @RequirePOST
+    public HttpResponse doStepsToJson(StaplerRequest req) {
+        Jenkins.getInstance().checkPermission(READ);
+
+        JSONObject result = new JSONObject();
+
+        String groovyAsString = req.getParameter("jenkinsfile");
+
+        try {
+            List<ModelASTStep> steps = Converter.scriptToPlainSteps(groovyAsString);
+            JSONArray array = new JSONArray();
+            for (ModelASTStep step : steps) {
+                array.add(step.toJSON());
+            }
+            result.accumulate("result", "success");
+            if (array.size() == 1) {
+                result.accumulate("json", array.get(0)); //TODO decide if consistently returning an array is better
+            } else {
+                result.accumulate("json", array);
+            }
+
+        } catch (MultipleCompilationErrorsException e) {
+            result.accumulate("result", "failure");
```

Comment:
At this point, we may want to abstract the error handling logic out, since it's basically the same for all `Jenkinsfile` -> JSON conversations (and all JSON -> `Jenkinsfile` conversions have the same error handling themselves).